src/garage/envs/environment.py

```diff
@@ -0,0 +1,176 @@
+"""Base Garage Environment API."""
+
+import abc
+
+
+class Environment(abc.ABC):
```

Comment:
Somewhere there should be a reference to the pickle protocol, and that it should work for this interface.